### PR TITLE
HW02 Mikhail Vorobev 

### PR DESCRIPTION
**Pull request by InversionSpaces**

### CODE REVIEW

File: recap/src/Imp/SLang.hs [hunk 1]
```diff
@@ -44,15 +74,85 @@ data SizedList v s where
...
+  return $ Write ec
+compile (TL.Seq e1 e2) i = do
+  e1c <- compile e1 i
+  e2c <- compile e2 i
+  return $ Seq geqRefl geqRefl e1c e2c
+compile (TL.If c t e) i = do
+  cc <- compile c i
+  tc <- compile t i
+  ec <- compile e i
+  return $ If geqRefl geqRefl geqRefl cc tc ec
```
Comment (kajigor) [inline recap/src/Imp/SLang.hs:152]: It's a matter of style, but by using `<*>` you can write this kind of code as a one-liner.

File: recap/src/Imp/SLang.hs [hunk 2]
```diff
@@ -19,23 +19,53 @@ data GEQ a b where
...
+    SafeExpr Int ci ->
+    SafeExpr a ti ->
+    SafeExpr a ei ->
+    SafeExpr a i
+  LetIn ::
+    GEQ i ei ->
+    SafeExpr Int ei ->
+    SafeExpr a (S i) ->
+    SafeExpr a i
+  Skip :: SafeExpr () i
```
Comment (kajigor) [inline recap/src/Imp/SLang.hs:68]: Why do you need a `GEQ` for each sub-expression?

File: recap/src/Imp/SLang.hs (hunk 1 above)
Comment (InversionSpaces) [reply recap/src/Imp/SLang.hs]: Used `<*>`

File: recap/src/Imp/SLang.hs (hunk 2 above)
Comment (InversionSpaces) [reply recap/src/Imp/SLang.hs]: Well my idea was to allow constructing expressions from subexpressions with another `i` parameter.
I guess with type inference it is unnecessary. I deleted this.

**Comment by kajigor** [comment]
Ok. Good job.

**Pull request closed without merging**